### PR TITLE
[LoongArch] Adjust LA64 data layout by using n32:64 in layout string

### PR DESCRIPTION
**Pull request by heiher**
Although i32 type is illegal in the backend, LA64 has pretty good support for i32 types by using W instructions.

By adding n32 to the DataLayout string, middle end optimizations will consider i32 to be a native type. One known effect of this is enabling LoopStrengthReduce on loops with i32 induction variables. This can be beneficial because C/C++ code often has loops with i32 induction variables due to the use of `int` or `unsigned int`.

If this patch exposes performance issues, those are better addressed by tuning LSR or other passes.

### CODE REVIEW

File: llvm/lib/IR/AutoUpgrade.cpp [hunk 1]
```diff
@@ -5368,8 +5368,8 @@ std::string llvm::UpgradeDataLayoutString(StringRef DL, StringRef TT) {
     return DL.empty() ? std::string("G1") : (DL + "-G1").str();
   }
 
-  if (T.isRISCV64()) {
-    // Make i32 a native type for 64-bit RISC-V.
+  if (T.isRISCV64() || T.isLoongArch64()) {
```
Comment (SixWeining) [inline llvm/lib/IR/AutoUpgrade.cpp:5371]: Better to sort alphabetically.

**Review by SixWeining — COMMENTED** [review]
@wangleiat What do you think?

**Comment by wangleiat** [comment]
> @wangleiat What do you think?

I think this is a good start, and we should make i32 truly legal on LoongArch64, just like RISC-V.

**Review by wangleiat — APPROVED** [review]
Thanks.

**Pull request merged**